csatar-plugins/forms/traits/AjaxControllerSimple.php

```diff
@@ -126,7 +125,8 @@ public function onSave() {
         }
 
         if (! $data = Input::get('data')) {
-            throw new ApplicationException("Error: The form could not be saved.");
+            $error = e(trans('csatar.forms::lang.errors.canNotSave'));
+            throw new ApplicationException($error);
```

Comment:
Ez a fájl megint tele van `ApplicationException`-nal.
Ez az eset különösen szót érdemel, mert a hibaüzenet nem a hibát írja le. Itt nem az a hiba, hogy nem tudtuk lementeni a form-ot, hanem az, hogy a `data` paraméter értéke üres (vagy nem is volt megadva egyáltalán).

Amúgy nincs valami validálás, amivel külön lehet kérni, hogy a data mező legyen kötelező? Hogy ne kelljen a kontrollert megtömni validálásokkal?

(Picit fennebb a 122-es sorban megint ott van bedrótozva, hogy a model név elé tegyünk oda egy `\`-t. Mi van ha már ott van? Megintcsak validálás...)

Reply:
Elméletileg el se juthat idáig úgy, hogy nincs `data` pont a validálás miatt, és a 122-ig se. Az eredetiben így volt, hát benne hagytam, szedjem ki?